raphtory/src/algorithms/algorithm_result.rs

```diff
@@ -55,32 +56,37 @@ pub struct AlgorithmRepr {
 ///
 /// This `AlgorithmResult` is returned for all algorithms that return a HashMap
 ///
-pub struct AlgorithmResult<K, V, O = V> {
+pub struct AlgorithmResult<G, V, O = V> {
```

Comment:
I believe there are a couple of things we need to fix in here:

- For all of the functions by default they should no longer return a string, instead they should return a vertex object. This doing a lot of extra work that we are 'not being asked for' and removes the actual planned functionality. 
- We should return to the old function names for the above, but add an `_names` variation which would map the vertex objects to their name/id
- We need to run some benchmarks on some larger graphs to make sure these changes aren't adding a load of overhead.